### ISSUE TITLE
Error "No servers configured" when using mastad0n's resolvers

### ISSUE THREAD

**Issue by caribpa** [DNSCrypt/dnscrypt-proxy#206]
Hi,
I've found that you still cannot use [mastad0n's resolvers](https://mastad0n.github.io/) due to "**No servers configured**" error. This issue is present in both stable (2.0.6-1) and git (2.0.6.r0.g4e671cf-1) versions.

I've tried to toggle all the secure settings, add prefixes to the sources, change the load balancing strategy, play with different resolvers and names for both the sources and the cache files, and so far the issue remains.

Here there are the .toml and the service log: [dnscrypt-report.zip](https://github.com/jedisct1/dnscrypt-proxy/files/1777575/dnscrypt-report.zip)

Maybe the way the resolvers.md file is processed is failing?

Thank you very much.

**Comment by ghost** [comment]
Let me know if there's anything I can do to help. 

I've had this same issue and fixed it doing wget on the toml file straight from the repository:

https://github.com/jedisct1/dnscrypt-proxy/blob/master/dnscrypt-proxy/example-dnscrypt-proxy.toml

Are you using Arch Linux by any chance?

I'm asking because I think there's something going wrong from the package at AUR.

**Comment by ghost** [comment]
@jedisct1 could this be due to dnscrypt-proxy not running as root anymore?

@Caribpa could you show us the output of:

> ls -ld /var/cache/private

In your logs it says dnscrypt-proxy is running as test-pc, maybe that folder is owned by a different user, like root?

**Comment by caribpa** [comment]
Bingo:
```sh
caribpa@test-pc /tmp % ls -ld /var/cache/private 
drwx------ 3 root root 28 mar  1 20:26 /var/cache/private
```

**Comment by ghost** [comment]
@Zirkelite could you help us?

I think something changed during updates because I've had the same issue before.

DynamicUser= is present in the latest systemd, and CacheDirectory= & LogsDirectory= are defined.

Yet I think some permission was left behind that makes it fail.

**Comment by Zirkelite** [comment]
@mastad0n Sorry for the late reply.

I have only experienced this issue once, and it was between two beta updates. I don't know what causes it, but when it happened to me I was able to solve it by deleting the public resolvers cache file.

**Comment by caribpa** [comment]
After trying many things, including @Zirkelite's suggestion, playing with /var/cache/private folder's permissions and the resolvers, nothing worked.
Some days after, I started to experiment with other settings in the .toml file and I found out the issue: **@mastad0n's resolvers won't work with require_nofilter = true**
That statement is a generalization because I haven't tried all the resolvers, only the USA, United Kingdom, Germany, Spain and Netherlands with everything in the .toml set to false and doh_servers = true & require_nofilter = true as follow:
```txt
# Use servers reachable over IPv4
ipv4_servers = true

# Use servers reachable over IPv6 -- Do not enable if you don't have IPv6 connectivity
ipv6_servers = false

# Use servers implementing the DNSCrypt protocol
dnscrypt_servers = false

# Use servers implementing the DNS-over-HTTPS protocol
doh_servers = true

## Require servers defined by remote sources to satisfy specific properties

# Server must support DNS security extensions (DNSSEC)
require_dnssec = false

# Server must not log user queries (declarative)
require_nolog = false

# Server must not enforce its own blacklist (for parental control, ads blocking...)
require_nofilter = true
```
After this setup, the [dnscrypt log](https://github.com/jedisct1/dnscrypt-proxy/files/1834237/dnscrypt-mastad0n-nofilter-true.txt) said "No servers configured", as you can see here:
```sh
● dnscrypt-proxy.service - DNSCrypt client proxy
   Loaded: loaded (/usr/lib/systemd/system/dnscrypt-proxy.service; disabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/dnscrypt-proxy.service.d
           └─override.conf
   Active: failed (Result: exit-code) since Wed 2018-03-21 16:25:41 WET; 2min 2s ago
     Docs: https://github.com/jedisct1/dnscrypt-proxy/wiki
  Process: 15972 ExecStart=/usr/bin/dnscrypt-proxy --config /etc/dnscrypt-proxy/dnscrypt-proxy.toml (code=exited, status=255)
 Main PID: 15972 (code=exited, status=255)

mar 21 16:25:41 test-pc dnscrypt-proxy[15972]: Source [https://mastad0n.github.io/resolvers/v2/usa.md] loaded
mar 21 16:25:41 test-pc dnscrypt-proxy[15972]: Source [https://mastad0n.github.io/resolvers/v2/germany.md] loaded
mar 21 16:25:41 test-pc dnscrypt-proxy[15972]: Source [https://mastad0n.github.io/resolvers/v2/netherlands.md] loaded
mar 21 16:25:41 test-pc dnscrypt-proxy[15972]: Source [https://mastad0n.github.io/resolvers/v2/portugal.md] loaded
mar 21 16:25:41 test-pc dnscrypt-proxy[15972]: No servers configured
mar 21 16:25:41 test-pc systemd[1]: dnscrypt-proxy.service: Main process exited, code=exited, status=255/n/a
mar 21 16:25:41 test-pc systemd[1]: dnscrypt-proxy.service: Failed with result 'exit-code'.
mar 21 16:25:41 test-pc systemd[1]: dnscrypt-proxy.service: Start request repeated too quickly.
mar 21 16:25:41 test-pc systemd[1]: dnscrypt-proxy.service: Failed with result 'exit-code'.
mar 21 16:25:41 test-pc systemd[1]: Failed to start DNSCrypt client proxy.

```

The only exception I found is the [anycast.md](https://mastad0n.github.io/resolvers/v2/anycast.txt) resolvers, so maybe there are some others but all @mastad0n's resolvers should be properly recognized.

If I use a secure and unfiltered configuration, as follow:
```txt
# Use servers reachable over IPv4
ipv4_servers = true

# Use servers reachable over IPv6 -- Do not enable if you don't have IPv6 connectivity
ipv6_servers = true

# Use servers implementing the DNSCrypt protocol
dnscrypt_servers = true

# Use servers implementing the DNS-over-HTTPS protocol
doh_servers = false

## Require servers defined by remote sources to satisfy specific properties

# Server must support DNS security extensions (DNSSEC)
require_dnssec = true

# Server must not log user queries (declarative)
require_nolog = true

# Server must not enforce its own blacklist (for parental control, ads blocking...)
require_nofilter = true
```
This is the [output of dnscrypt ](https://github.com/jedisct1/dnscrypt-proxy/files/1834292/dnscrypt-servers-public-resolvers-nofilter-true.txt) when using [public-resolvers.md](https://download.dnscrypt.info/dnscrypt-resolvers/v2/public-resolvers.md):
```sh
mar 21 16:05:14 test-pc dnscrypt-proxy[15311]: [qag.me] OK (crypto v2) - rtt: 189ms
mar 21 16:05:14 test-pc dnscrypt-proxy[15311]: [scaleway-fr] OK (crypto v2) - rtt: 100ms
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: [2.dnscrypt-cert.securedns.eu.] the key validity period for this server is excessively long (365 days), significantly reducing reliability and forward security.
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: [securedns] OK (crypto v1) - rtt: 71ms
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: [2.dnscrypt-cert.securedns.eu.] the key validity period for this server is excessively long (365 days), significantly reducing reliability and forward security.
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: [securedns-ipv6] OK (crypto v1) - rtt: 71ms
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: [soltysiak] OK (crypto v1) - rtt: 88ms
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: [ventricle.us] OK (crypto v1) - rtt: 149ms
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: Server with the lowest initial latency: flatty.co (rtt: 63ms)
mar 21 16:05:15 test-pc dnscrypt-proxy[15311]: dnscrypt-proxy is ready - live servers: 30
```

And this is the [output of dnscrypt](https://github.com/jedisct1/dnscrypt-proxy/files/1834310/dnscrypt-servers-netherlands-nofilter-false.txt) when using @mastad0n's [netherlands.md](https://mastad0n.github.io/resolvers/v2/netherlands.txt) with **require_nofilter = false**:
```sh
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [dnscrypt.eu-nl] OK (crypto v1) - rtt: 69ms
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [dnscrypt.nl-ns0] OK (crypto v1) - rtt: 71ms
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [2.dnscrypt-cert.securedns.eu.] the key validity period for this server is excessively long (365 days), significantly reducing reliability and forward security.
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [securedns] OK (crypto v1) - rtt: 104ms
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [2.dnscrypt-cert.securedns.eu.] the key validity period for this server is excessively long (365 days), significantly reducing reliability and forward security.
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [securedns-ipv6] OK (crypto v1) - rtt: 71ms
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [dnscrypt.nl-ns0-ipv6] OK (crypto v1) - rtt: 101ms
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: [d0wn-nl-ns4] OK (crypto v1) - rtt: 69ms
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: Server with the lowest initial latency: dnscrypt.eu-nl (rtt: 69ms)
mar 21 16:03:32 test-pc dnscrypt-proxy[15245]: dnscrypt-proxy is ready - live servers: 6
```

As you can see they have in common these servers:
```sh
[2.dnscrypt-cert.securedns.eu.]
[securedns]
[2.dnscrypt-cert.securedns.eu.]
[securedns-ipv6]
```
And [securedns.eu](https://securedns.eu/)'s servers are [uncensored](https://github.com/DNSCrypt/dnscrypt-resolvers/blob/master/v2/public-resolvers.md#securedns), which proves the issue.
The same applies to other servers like [dnscrypt.nl](https://dnscrypt.nl/) also marked as [uncensored](https://github.com/DNSCrypt/dnscrypt-resolvers/blob/master/v2/public-resolvers.md#dnscryptnl-ns0).

**Comment by caribpa** [comment]
I don't know why this issue is closed as it is still present in 2.0.8

**Comment by jedisct1** [comment]
@mastad0n 's lists really need an update. Servers are missing, but I also think that the stamps are not up to date.
